src/library/vm/vm.cpp

```diff
@@ -1371,14 +1472,49 @@ environment add_vm_code(environment const & env, name const & fn, unsigned arity
     return update_vm_code(new_env, fn, code_sz, code, args_info, pos);
 }
 
-optional<vm_decl> get_vm_decl(environment const & env, name const & n) {
+optional<vm_decl> get_vm_override_decl(environment const & env, vm_decl const & decl) {
+    vm_decls const & ext = get_extension(env);
+    if (!get_vm_override_enabled(get_global_ios().get_options()))
```

Comment:
```suggestion
    if (!get_vm_override_enabled(opts))
```
And pass the `m_options` field from `vm_state` as an argument.